### PR TITLE
feat(i18n): convert document actions strings to use i18n primitives

### PR DESCRIPTION
**Pull request by bjoerge**
### Description
This converts various document action strings into 18n-keys

Also includes some minor cleanup. Commits can be reviewed one-by-one.

### CODE REVIEW

**Review by binoy14 — APPROVED** [review]

File: packages/sanity/src/desk/i18n/resources.ts [hunk 1]
```diff
@@ -37,11 +41,80 @@ const deskLocaleStrings = {
...
+  /** Fallback tooltip for the Unpublish document action when publish is invoked for a document with live edit enabled.*/
+  'action.unpublish.liveEdit.disabled':
+    'This document has live edit enabled and cannot be unpublished',
+
+  /** --- RESTORE ACTION --- */
+  /** Label for the "Restore" document action */
+  'action.restore.label': 'Restore',
+
+  /** Fallback tooltip for when user is looking at the initial version */
+  'action.restore.disabled.cantRestoreInitial': "You can't restore to the initial version",
```
Comment (rexxars) [inline packages/sanity/src/desk/i18n/resources.ts:110]: I suggest we do not use contractions in identifiers, so I would write this as `action.restore.disabled.cannot-restore-initial`

**Review by rexxars — CHANGES_REQUESTED** [review]
We should align on camelCase vs kebab-case for the i18n resources, eg `action.delete.disabled.notReady` vs `action.delete.disabled.not-ready`.

I am personally leaning towards kebab-case, but open to input.

File: packages/sanity/src/desk/i18n/resources.ts (hunk 1 above)
Comment (bjoerge) [reply packages/sanity/src/desk/i18n/resources.ts]: oh, good point - guess we should stick to snake-case as well. Fixed in 8dde36dee0e58072869656ba530a8f8c62286004

**Comment by bjoerge** [comment]
> I am personally leaning towards kebab-case, but open to input.

Agree. Unless we think these will appear as object keys I don't see any point of using camel case

**Review by rexxars — CHANGES_REQUESTED** [review]
Since we agree, it would be great if you changed the keys here to be kebabcase, so `action.restore.confirmDialog.confirmDiscardChanges` => `action.restore.confirmDialog.confirm-discard-changes` etc.

**Comment by bjoerge** [comment]
> Since we agree, it would be great if you changed the keys here to be kebabcase, so action.restore.confirmDialog.confirmDiscardChanges => action.restore.confirmDialog.confirm-discard-changes etc. 

Fixed in c07fd19 - thanks for pointing out!

**Review by rexxars — CHANGES_REQUESTED** [review]
Still seems to be using camelcase? Examples:

- discardChanges => discard-changes
- confirmDialog => confirm-dialog
- notPublished => not-published
- notReady => not-ready
- nothingToDelete => nothing-to-delete

There's more, but you get the idea

**Comment by bjoerge** [comment]
🤦🏼

aight, new attempt. Also kebab-cased some existing keys for the publish action.

**Review by rexxars — APPROVED** [review]

**Pull request merged**